src/libsyncengine/jobs/network/upload_session/uploadsession.cpp

```diff
@@ -285,62 +285,72 @@ bool UploadSession::sendChunks() {
                 LOGW_WARN(_logger, L"Failed to read chunk - path=" << Path2WStr(_filePath).c_str());
                 readError = true;
                 break;
-            } else {
-                std::streamsize actualChunkSize = file.gcount();
-                std::string chunkContent = std::string(memblock.get(), actualChunkSize);
-
-                std::shared_ptr<UploadSessionChunkJob> chunkJob = nullptr;
-                try {
-                    chunkJob = std::make_shared<UploadSessionChunkJob>(_driveDbId, _filePath, _sessionToken, chunkContent,
-                                                                       chunkNb, actualChunkSize, jobId());
-                } catch (std::exception const &e) {
-                    LOG_ERROR(_logger, "Error in UploadSessionChunkJob::UploadSessionChunkJob: " << e.what());
-                    jobCreationError = true;
-                    break;
-                }
+            }
 
-                if (XXH3_64bits_update(state, chunkJob->chunkHash().data(), chunkJob->chunkHash().length()) == XXH_ERROR) {
-                    LOGW_WARN(_logger,
-                              L"Checksum computation " << jobId() << L" failed for file " << Path2WStr(_filePath).c_str());
-                    checksumError = true;
-                    break;
-                }
+            std::streamsize actualChunkSize = file.gcount();
+            std::string chunkContent = std::string(memblock.get(), actualChunkSize);
```

Comment:
Could this be moved after the conditional `if (actualChunkSize <= 0)`?

Reply:
Yes, you are right, it's done